### ISSUE TITLE
Blobs not spinning even though stage is still running

### ISSUE THREAD

**Issue by chrissie-c** [jenkinsci/pipeline-graph-view-plugin#369]
### Jenkins and plugins versions report

<details>
  <summary>Environment</summary>

  
  ```text
  Paste the output here
  ```

</details>

### What Operating System are you using (both controller, and any agents involved in the problem)?

```
Jenkins: 2.449
OS: Linux - 6.1.0-18-amd64
Java: 17.0.10 - Debian (OpenJDK 64-Bit Server VM)
---
PrioritySorter:5.1.0
ant:497.v94e7d9fffa_b_9
antisamy-markup-formatter:162.v0e6ec0fcfcf6
apache-httpcomponents-client-4-api:4.5.14-208.v438351942757
asm-api:9.6-3.v2e1fa_b_338cd7
authentication-tokens:1.53.v1c90fd9191a_b_
bootstrap5-api:5.3.3-1
bouncycastle-api:2.30.1.77-225.v26ea_c9455fd9
branch-api:2.1152.v6f101e97dd77
build-timeout:1.32
caffeine-api:3.1.8-133.v17b_1ff2e0599
checks-api:2.0.2
cloudbees-bitbucket-branch-source:880.vcf4056c5a_71f
cloudbees-folder:6.928.v7c780211d66e
command-launcher:107.v773860566e2e
commons-lang3-api:3.13.0-62.v7d18e55f51e2
commons-text-api:1.11.0-95.v22a_d30ee5d36
credentials:1337.v60b_d7b_c7b_c9f
credentials-binding:657.v2b_19db_7d6e6d
data-tables-api:2.0.2-1
depgraph-view:1.0.5
display-url-api:2.200.vb_9327d658781
durable-task:550.v0930093c4b_a_6
echarts-api:5.5.0-1
favorite:2.208.v91d65b_7792a_c
font-awesome-api:6.5.1-3
git:5.2.1
git-client:4.7.0
github:1.38.0
github-api:1.318-461.v7a_c09c9fa_d63
github-branch-source:1781.va_153cda_09d1b_
github-oauth:597.ve0c3480fcb_d0
gradle:2.10
gson-api:2.10.1-15.v0d99f670e0a_7
handy-uri-templates-2-api:2.1.8-30.v7e777411b_148
htmlpublisher:1.33
http_request:1.18
instance-identity:185.v303dc7c645f9
ionicons-api:56.v1b_1c8c49374e
jackson2-api:2.16.2-378.v7e79818f53ce
jakarta-activation-api:2.1.3-1
jakarta-mail-api:2.1.3-1
javax-activation-api:1.2.0-6
javax-mail-api:1.6.2-9
jaxb:2.3.9-1
jdk-tool:73.vddf737284550
jfrog:1.5.0
jjwt-api:0.11.5-77.v646c772fddb_0
joda-time-api:2.12.7-29.v5a_b_e3a_82269a_
jquery3-api:3.7.1-2
json-api:20240303-41.v94e11e6de726
json-path-api:2.9.0-52.v57de85cc4722
junit:1259.v65ffcef24a_88
ldap:711.vb_d1a_491714dc
lockable-resources:1243.v346d600eea_24
logging:1.0.0
mailer:472.vf7c289a_4b_420
matrix-auth:3.2.2
matrix-project:822.824.v14451b_c0fd42
metrics:4.2.21-449.v6960d7c54c69
mina-sshd-api-common:2.12.0-90.v9f7fb_9fa_3d3b_
mina-sshd-api-core:2.12.0-90.v9f7fb_9fa_3d3b_
okhttp-api:4.11.0-172.vda_da_1feeb_c6e
pam-auth:1.10
pipeline-build-step:540.vb_e8849e1a_b_d8
pipeline-github-lib:42.v0739460cda_c4
pipeline-graph-analysis:216.vfd8b_ece330ca_
pipeline-graph-view:231.vecb_8c5841335
pipeline-groovy-lib:704.vc58b_8890a_384
pipeline-input-step:491.vb_07d21da_1a_fb_
pipeline-milestone-step:111.v449306f708b_7
pipeline-model-api:2.2184.v0b_358b_953e69
pipeline-model-definition:2.2184.v0b_358b_953e69
pipeline-model-extensions:2.2184.v0b_358b_953e69
pipeline-rest-api:2.34
pipeline-stage-step:305.ve96d0205c1c6
pipeline-stage-tags-metadata:2.2184.v0b_358b_953e69
pipeline-stage-view:2.34
plain-credentials:179.vc5cb_98f6db_38
plugin-util-api:4.1.0
pubsub-light:1.18
remote-file:1.24
resource-disposer:0.23
scm-api:689.v237b_6d3a_ef7f
script-security:1326.vdb_c154de8669
snakeyaml-api:2.2-111.vc6598e30cc65
sse-gateway:1.26
ssh-credentials:322.v124df57ed808
ssh-slaves:2.948.vb_8050d697fec
sshd:3.322.v159e91f6a_550
structs:337.v1b_04ea_4df7c8
timestamper:1.26
token-macro:400.v35420b_922dcb_
trilead-api:2.142.v748523a_76693
variant:60.v7290fc0eb_b_cd
workflow-aggregator:596.v8c21c963d92d
workflow-api:1291.v51fd2a_625da_7
workflow-basic-steps:1049.v257a_e6b_30fb_d
workflow-cps:3883.vb_3ff2a_e3eea_f
workflow-durable-task-step:1331.vc8c2fed35334
workflow-job:1400.v7fd111b_ec82f
workflow-multibranch:783.va_6eb_ef636fb_d
workflow-scm-step:427.v4ca_6512e7df1
workflow-step-api:657.v03b_e8115821b_
workflow-support:865.v43e78cc44e0d
ws-cleanup:0.45
```

### Reproduction steps

Create a pipeline task from the following script:

```groovy
def waitrand(num) {
    stage("Waiting ${num}") {
        try {
          def delay = new Random().nextInt(60)
          sh (returnStdout: true, script: "for i in `seq 1 ${delay}`; do sleep 1; echo iter \$i; done")
          println("finshed sleeping")
        } catch (err) {
            println("Error caught "+err)
        }
    }
}

def waitjob() {

    waitrand(1)
    waitrand(2)
    waitrand(3)
}

def jobs = [:]

pipeline {
    agent any
    stages {
        stage("prep") {
            steps {
                echo "prep step"
            }
        }
      stage("run") {
        steps {
       	  script {
            jobs['one'] = {waitjob()}
            jobs['two'] = {waitjob()}
            jobs['three'] = {waitjob()}
            jobs['four'] = {waitjob()}
            jobs['five'] = {waitjob()}
            jobs['six'] = {waitjob()}
            try {
                stage("first") {
                    parallel jobs
                }
            } catch (err) {
                println("Error in parallel1")
            }
            stage("second") {
                try {
                    parallel jobs
                } catch(err) {
                    println("Error in parallel2")
                }
            }
          }
       	}
      }
    }
    post {
      always {
        echo "finishing"
      }
    }
}
```
Then run it and watch the pipeline overview (graph) page.

### Expected Results

I would expect the 'blobs' to keep spinning blue until the stage had completed and then turn green or red.

### Actual Results

Running the attached pipeline shows the stages incorrectly. Even though some stages are still running, the stage 'blobs' turned green while the script stages are still running. 
![Screenshot from 2024-03-18 10-52-46](https://github.com/jenkinsci/pipeline-graph-view-plugin/assets/3415721/ced712fe-3678-4871-bead-fb5b4a9df7b1)

### Anything else?

This seems to be a new bug since  v232 (I think) as it didn't happen on older versions.

### Are you interested in contributing a fix?

I'm not sure I am able to help, but I'm happy to test potential fixes on a test system

**Comment by galindro** [comment]
Try to upgrade to [232.vc7ca_8d934725](https://github.com/jenkinsci/pipeline-graph-view-plugin/releases/tag/232.vc7ca_8d934725) please and test again.

**Comment by timbrown5** [comment]
Your environment paste says you are using.
> `pipeline-graph-view:231.vecb_8c5841335`
But you say this was introduced in 232.
> This seems to be a new bug since v232 (I think) as it didn't happen on older versions.

Could you double check which version you are seeing this issue. If it's before 232 then, as @galindro suggests, please try with 232 - that has a fix for incorrect status and timings in it.

**Comment by chrissie-c** [comment]
Thanks for the reply.

its odd, I checked both my instances and they are both running Version 232.vc7ca_8d934725 . I only started seeing this behaviour when the plugin got upgraded so I can't be 100% sure that the 'working' version was 231, but I definitely had one jenkins instance that worked and one that didn't. Now they are both on 232 then both are showing the missing spinners. I hope that's clear, sorry for confusion.
